File: cii/tests/word_count.py
#!/usr/bin/env python3

import os
import sys


def main():
    filename = sys.argv[1]
    words = dict()
    with open(filename) as fh:
        for line in fh.readlines():
            for word in line.split():
                # if not word[-1].isalpha():
                #     word = word[:-1]
                count = words.get(word, 0)
                words[word] = count + 1

    for item in sorted(words.keys()):
        print(item, "occurs", words[item], "times")


if __name__ == "__main__":
    main()

File: cii/tests/word_count.rs
use std::collections::BTreeMap;
use std::env;
use std::fs::File;
use std::io::{self, BufRead, BufReader};

fn word_count(filename: &str) -> Result<(), io::Error> {
    let fd = File::open(filename)?;
    let reader = BufReader::new(fd);
    let mut counts = BTreeMap::<String, usize>::new();

    for line in reader.lines() {
        let line = line?;
        for word in line.split_ascii_whitespace() {
            if let Some(count) = counts.get_mut(word) {
                *count += 1;
            } else {
                counts.insert(word.to_owned(), 1);
            }
        }
    }

    for (key, value) in &counts {
        println!("{key}: {value}");
    }

    Ok(())
}

fn main() {
    for arg in env::args().skip(1) {
        word_count(&arg).unwrap();
    }
}
